src/core/quadFeature.js

```diff
@@ -0,0 +1,443 @@
+//////////////////////////////////////////////////////////////////////////////
+/**
+ * Create a new instance of class quadFeature
+ *
+ * @class
+ * @param {Object} arg Options object
+ * @extends geo.feature
+ * @param {Object|string|Function} [color] Color for quads without images.
+ *   Default is white ({r: 1, g: 1, b: 1}).
+ * @param {number|Function} [opacity=1] Opacity for quad
+ * @param {number|Function} [depth=0] Default z-coordinate for positions that
+ *   don't explicitly specify one.
+ * @param {boolean|Function} [drawOnAsyncResourceLoaded=true] Redraw quads
+ *   when images are loaded after initial render.
+ * @param {Image|string|Function} [image] Image for each data item.  If
+ *   undefined or null, the quad is a solid color.  Default is (data).image.
+ * @param {Object|string|Function} [previewColor=null] If specified, a color to
+ *   show on image quads while waiting for the image to load.
+ * @param {Image|string|Function} [previewImage=null] If specified, an image to
+ *   show on image quads while waiting for the quad-specific image to load.
+ *   This will only be shown if it is already loaded.
+ * @param {Object|Function} [position] Position of the quad.  Default is
+ *   (data).  The position is an Object which specifies the corners of the
+ *   quad: ll, lr, ur, ul.  At least two opposite corners must be specified.
+ *   The corners do not have to physically correspond to the order specified,
+ *   but rather correspond to that part of an image (if there is one).  If a
+ *   corner is unspecified, it will use the x coordinate from one adjacent
+ *   corner, the y coordinate from the other adjacent corner, and the average
+ *   z value of those two corners.  For instance, if ul is unspecified, it is
+ *   {x: ll.x, y: ur.y}.  Note that each quad is rendered as a pair of
+ *   triangles: (ll, lr, ul) and (ur, ul, lr).  Nothing special is done for
+ *   quads that are no convex or quads that have substantially different
```

Comment:
quads that are no convex --> I haven't see "no convex" usage in long time but it is valid. Just wondering if you have used it purposely.